group.py

```diff
@@ -11,11 +11,97 @@ def _init_(self, name, age, occupation ):
         self.occupation = occupation
         self.relationships = {}  #define relationship
 
-    def add_relationships(self, name, relationships):
+    #def add_relationships(self, name, relationships):
         
 
 
-Jill_relations = {Zalika: ""friend", John: "partner"}
-    
+Jill_relations = {"Zalika": "friend", "John": "partner"}
+
+
+#Homework: Extension
+#TODO: 
+#Add some code that makes use of comprehension expressions to your group.py file so that it prints out the following when the script is run:
+
+# 1. the maximum age of people in the group
+# 2. the average (mean) number of relations among members of the group
+# 3. the maximum age of people in the group that have at least one relation
+# 4. [more advanced] the maximum age of people in the group that have at least one friend
+
+# Let's use the dictionary approach for this 
+
+group = {
+    "Jill": {
+        "age": 26,
+        "job": "biologist",
+        "relations": {
+            "Zalika": "friend",
+            "John": "partner"
+        }
+    },
+    "Zalika": {
+        "age": 28,
+        "job": "artist",
+        "relations": {
+            "Jill": "friend"
+        }
+    },
+    "John": {
+        "age": 27,
+        "job": "writer",
+        "relations": {
+            "Jill": "partner"
+        }
+    },
+    "Nash": {
+        "age": 34,
+        "job": "chef",
+        "relations": {
+            "John": "cousin",
+            "Zalika": "landlord"
+        }
+    }
+}
+
+#1. The maximum age of people in the group 
+
+max_age = max([group[person]["age"] for person in group])
```

Comment:
You can also iterate just through the values in a dictionary
```suggestion
max_age = max([person["age"] for person in group.values()])
```